resources/metadata/versioning.yaml

```diff
@@ -9,17 +9,41 @@ metadata:
     Prepares and potentially updates the artifact's version before building the artifact.
 
     The continuous delivery process requires that each build is done with a unique version number.
+    There are two common patterns found:
 
-    The version generated using this step will contain:
+    ### 1. Continuous Deployment pattern with automatic versioning
 
-    * Version (major.minor.patch) from descriptor file in master repository is preserved. Developers should be able to autonomously decide on increasing either part of this version number.
-    * Timestamp
-    * CommitId (by default the long version of the hash)
+    The team has full authority on `<major>.<minor>.<patch>` and can increase any part whenever required.
+    Nonetheless, the automatic versioning makes sure that every build will create a unique version by appending `<major>.<minor>.<patch>` with a buildversion (we use a timestamp) and optinally the commitId.
 
-    Optionally, but enabled by default, the new version is pushed as a new tag into the source code repository (e.g. GitHub).
-    If this option is chosen, git credentials and the repository URL needs to be provided.
-    Since you might not want to configure the git credentials in Jenkins, committing and pushing can be disabled using the `commitVersion` parameter as described below.
-    If you require strict reproducibility of your builds, this should be used.
+    In order to represent this version also in the version control system the new unique version will be pushed with a dedicated tag (`<tagPrefix><major>.<minor>.<patch><unique extension>`).
+
+    Depending on the build tool used and thus the allowed versioning format the `<unique extension>` varies.
+
+    **Remarks:**
+
+    * There is no commit to master since this would create a perpetuum mobile and just trigger the next automatic build with automatic versioning, and so on ...
```

Comment:
I think this is clear and can be omitted.

Reply:
I am not sure about that. I keep telling this to people ...